examples/platform/silabs/SiWx917/SiWx917/hal/rsi_hal_mcu_m4.c

```diff
@@ -0,0 +1,267 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "rsi_board.h"
+#include "rsi_driver.h"
+#include "rsi_m4.h"
+#ifdef COMMON_FLASH_EN
+#include "rsi_power_save.h"
+#endif
+
+rsi_m4ta_desc_t tx_desc[2];
+rsi_m4ta_desc_t rx_desc[2];
+
+uint32_t NVIC_GetIRQEnable(IRQn_Type IRQn)
+{
+    return ((NVIC->ICER[((uint32_t)(IRQn) >> 5)] & (1 << ((uint32_t)(IRQn) &0x1F))) ? 1 : 0);
+}
+
+void rsi_assertion(uint16_t assertion_val, const char * string)
+{
+    uint16_t i;
+    if (assertion_val == 0)
+    {
+        for (i = 0; i < strlen(string); i++)
```

Comment:
why loop if 
DEBUG_UART and  DEBUG_ASSERTION aren't define

Same for the other loop.

In general, this function has alot of line for no reason.

```
void rsi_assertion(uint16_t assertion_val, const char * string)
{
#ifdef DEBUG_UART
#ifdef DEBUG_ASSERTION
    for (int16_t i = 0; i < strlen(string); i++)
    {
        Board_UARTPutChar(string[i]);
    }
#endif
#endif

    if (assertion_val != 0)
    {

        while (1);
    }
}
```
I suggest using something like this. Easier to read, no code duplication or unused loops

Reply:
removed the file which are being used directly from the submodule
"${wiseconnect_sdk_root}/platforms/si91x/hal/src/rsi_bootup_config.c", "${wiseconnect_sdk_root}/platforms/si91x/hal/src/rsi_hal_mcu_m4.c", "${wiseconnect_sdk_root}/platforms/si91x/hal/src/rsi_hal_mcu_m4_rom.c", "${wiseconnect_sdk_root}/platforms/si91x/hal/src/rsi_hal_mcu_interrupt.c",